client/src/app/login.service.ts

```diff
@@ -15,9 +15,12 @@ export class LoginService {
   ) {}
 
   loginWithGitHub() {
-    this.authService
-      .linkOrSignInWithGitHub()
-      .pipe(switchMap(loginUser => this.userService.createUserIfMissing()))
+    this.userService
+      .getCurrentUserSaved()
+      .pipe(
+        switchMap(saved => (saved ? this.authService.linkOrSignInWithGitHub() : this.authService.signInWithGitHub())),
```

Comment:
While this looks like what we want, it actually turns out this line introduces a regression in our app, plus, I'm unsure if `linkOrSignInWithGitHub()` will ever be called in practice. Let me elaborate on both points:

- **Regression**

  This is something I forgot to mention in the original issue (#805), but one of the reasons we've (so far) always tried to first **link** the current user, is because we want to ensure that data that has been created by `currentUser` is going to be transferred to the `authUser`, once signed up. In other words, when you go to MachineLabs as an `anonymousUser` (everyone who's not logged in, is that by default), and you create labs, once you **do** login, we actually preserve all the labs you've created so far as an anonymous user. That's because we're linking the anonymous user we know at that point, with the auth user that is created from the provider auth data.

  Going with `signInWithGitHub()` right away, means we don't link and simply create a completely different user, which would mean, if you want to keep your labs you've created, you'll have to fork those as auth user from that anonymous user you've been before.

  I think this is also why this issue turns out to be trickier than expected. Because what we want is to only link if we know the user that wants to authenticate hasn't authenticated before. This is what I meant in the original issue:

  > We already have this /users node, but we'd still need to figure out for what users to check against. Obviously, that'd be for example the github provider data email address, but that one we're getting from the first popup that we're trying to get rid off. Happy to discuss ideas here.


  Now the question would be: how can we know whether an anonymous user has authenticated before? We would only know if the anonymous user tells us the email address that he/she uses with the provider she wants to authenticate with, so we can check whether that exists in `/users`. That's kind of what `linkOrSignInWithGitHub()` does, because it performs the OAuth popup with the provider (GitHub), which authenticates the user. Unfortunately, the method throws if the auth user does in fact exist, which is why we catch the error and simply perform a normal signIn (which causes the second popup).

  It's a little bit of a chicken and egg problem, and **the more I think about it, the more I'm leaning towards dropping the linking all together, which brings me to point 2**.

- **`linkOrSignInWithGithub()` will probably never be called anyway**

  As it is right now in this PR, we only call `linkOrSignInWithGitHub()`, if the current user does exists in our `/users` node, or in other words, if we know the user has authenticated before. `getCurrentUserSaved()` checks that by using the `currentUser` id. Since we know `currentUser` has to be an `anonymous` user at this point, it will actually never find it in `/users`. That's because anonymous users always get their own uid from firebase and are therefore always different users. If you're logged into MachineLabs and you logout, you become an anonymous user again with a brand new id. 

  This means in practice, we actually always call `signInWithGitHub()` no matter what and linking never takes place.

Now, having all that said, as mentioned I'm actually leaning towards to dropping the linking all together. In practice, anonymous users won't create tons of different labs anyways until they can actually execute them (for that you need to be an auth user). So we might not actually miss out so much if we can't preserve labs created by anonymous users.

@myspivey I hope this all makes sense and I'm curious to hear what you think about this. Maybe you actually do have an idea how to solve the chicken and egg problem in a feasible way.